app/uk/gov/hmrc/agentsubscription/connectors/AuthConnector.scala

```diff
@@ -16,40 +16,88 @@
-import scala.language.postfixOps
-import uk.gov.hmrc.http.{ HeaderCarrier, HttpGet, Upstream4xxResponse }
-
-@Singleton
-class AuthConnector @Inject() (@Named("auth-baseUrl") baseUrl: URL, httpGet: HttpGet) {
-  val authorityUrl = new URL(baseUrl, "/auth/authority")
-  def currentAuthority()(implicit hc: HeaderCarrier, ec: ExecutionContext): Future[Option[Authority]] = {
-    val response: Future[JsValue] = httpGet.GET[JsValue](authorityUrl.toString)
-    response flatMap { r =>
-      for {
-        userDetails <- userDetails(authorityUrl, (r \ "userDetailsLink").as[String])
-        enrolmentsUrl <- Future successful (r \ "enrolments").as[String]
-      } yield Some(Authority(authorityUrl, userDetails.authProviderId, userDetails.authProviderType, userDetails.affinityGroup, enrolmentsUrl))
-    } recover {
-      case error: Upstream4xxResponse if error.upstreamResponseCode == 401 => None
-      case e => throw e
-    }
-  }
+import play.api.mvc.{Result, _}
+import uk.gov.hmrc.agent.kenshoo.monitoring.HttpAPIMonitor
+import uk.gov.hmrc.agentsubscription.MicroserviceAuthConnector
+import uk.gov.hmrc.agentsubscription.controllers.ErrorResult._
+import uk.gov.hmrc.auth.core
+import uk.gov.hmrc.auth.core.AuthProvider.GovernmentGateway
+import uk.gov.hmrc.auth.core.retrieve.Retrievals.{affinityGroup, allEnrolments, credentials}
+import uk.gov.hmrc.auth.core.retrieve.{Credentials, Retrieval, ~}
+import uk.gov.hmrc.auth.core.{Enrolment, _}
+import uk.gov.hmrc.http.HeaderCarrier
+import uk.gov.hmrc.play.microservice.controller.BaseController
+
+import scala.concurrent.ExecutionContext.Implicits.global
+import scala.concurrent.Future
+
+case class Provider(providerId: String, providerType: String)
+
+class AuthConnector @Inject()(metrics: Metrics, microserviceAuthConnector: MicroserviceAuthConnector)
+  extends HttpAPIMonitor with AuthorisedFunctions with BaseController {
+  override def authConnector: core.AuthConnector = microserviceAuthConnector
+
+  override val kenshooRegistry = metrics.defaultRegistry
+  private type SubscriptionAuthAction = Request[JsValue] => Future[Result]
+  private type RegistrationAuthAction = Request[AnyContent] => Provider => Future[Result]
+
+  private val affinityGroupAllEnrols: Retrieval[~[Option[AffinityGroup], Enrolments]] = affinityGroup and allEnrolments
 
-  private def userDetails(authorityUrl: URL, userDetailsLink: String)(implicit hc: HeaderCarrier, ec: ExecutionContext): Future[UserDetails] = {
-    val absoluteUserDetailsUrl = new URL(authorityUrl, userDetailsLink).toString
-    httpGet.GET[UserDetails](absoluteUserDetailsUrl)
+  private val AuthProvider: AuthProviders = AuthProviders(GovernmentGateway)
+  private val agentEnrol = "HMRC-AS-AGENT"
+  private val agentEnrolId = "AgentReferenceNumber"
+  private val isAnAgent = true
+
+  implicit val hc: HeaderCarrier = new HeaderCarrier
+
+  def forSubscription(action: SubscriptionAuthAction) = Action.async(parse.json) {
```

Comment:
IMHO this is not a good idea to mix connector stuff with auth actions, better to keep them separate as before, WDYT?